packages/edit-site/src/components/sidebar/template-card/template-areas.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import {
+	Button,
+	__experimentalHeading as Heading,
+} from '@wordpress/components';
+import { getTemplatePartIcon } from '@wordpress/editor';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { store as editSiteStore } from '../../../store';
+import { TEMPLATE_PART_AREA_TO_NAME } from '../../../store/constants';
+
+function TemplateAreaItem( { area, clientId } ) {
+	const { selectBlock, toggleBlockHighlight } = useDispatch(
+		blockEditorStore
+	);
+	const highlightBlock = () => toggleBlockHighlight( clientId, true );
+	const cancelHighlightBlock = () => toggleBlockHighlight( clientId, false );
+
+	return (
+		<Button
+			className="edit-site-template-card__template-areas-item"
+			icon={ getTemplatePartIcon( area ) }
+			onMouseOver={ highlightBlock }
+			onMouseLeave={ cancelHighlightBlock }
+			onFocus={ highlightBlock }
+			onBlur={ cancelHighlightBlock }
+			onClick={ () => {
+				selectBlock( clientId );
+			} }
+		>
+			{ TEMPLATE_PART_AREA_TO_NAME[ area ] }
+		</Button>
+	);
+}
+
+export default function TemplateAreas() {
```

Comment:
I see a similar `TemplateAreas` component in https://github.com/WordPress/gutenberg/pull/35202. Can we re-use?

Reply:
Probably not a good idea, since that one is using `<MenuItem>` and this is using plain `<Button>`. I've already moved some common parts to a selector, the rests are mostly UI. We can create a UI component for them if there's a need.